Documentation/GettingStarted.md

```diff
@@ -0,0 +1,373 @@
+
+```json
+{
+  "kind": "song",
+  "artistId": 5448756,
+  "collectionId": 719245563,
+  "trackId": 719245998,
+  "artistName": "The Long Winters",
+  "collectionName": "Ultimatum - EP",
+  "trackName": "The Commander Thinks Aloud",
+  "previewUrl": "http://audio.itunes.apple.com/apple-assets-us-std-000001/<snipped>/.m4a",
+  "artworkUrl100": "http://is2.mzstatic.com/image/thumb/<snipped>/100x100bb.jpg",
+  "trackTimeMillis": 326053,
+  "trackNumber": 1,
+  "trackCount": 6,
+  // ...
+}
+```
+
+Representing this data in Swift would look like this:
+
+```swift
+struct Song {
+  var artistId: Int
+  var collectionId: Int
+  var trackId: Int
+  var artistName: String
+  var collectionName: String
+  var trackName: String
+  var preview: URL
+  var artwork: URL?
+  var durationMilliseconds: Int
+  var trackNumber: Int?
+  var trackCount: Int?
+}
+```
+
+Your model object doesn't have to come from JSON, it could come from Core Data, SQLite, or any other source. 
+
+> How to serialize & deserialize from your data source into a model struct like the above example is beyond the scope of this document.
+
+> The example application happens to add an initializer to the `Song` struct to inflate from JSON, along with a protocol to make that common across the other model types returned from iTunes. If interested, you can see that [here](../Examples/iTunesSearch/Shared/ModelSerialization.swift).
+
+> - [ ] TODO for @wkiefer: Replace this with `Codable` in Swift 4.
+
+Conformance to `Model` is quite simple, as there are only two properties to implement:
+
+- `modelId` is a string that guarantees uniqueness for that model object across your application.
+- `modelVersion` is a hash that represents the version of that object and lets Pilot know if important model data has changed.
+
+These two properties are used to provide automated delta updates across any heterogeneous collection of model objects. We'll see more about that later one.
+
+For our `Song`, we can conform to `Model` with the following code:
+
+```swift
+extension Song: Model {
+    
+  var modelId: ModelId { return String(trackId) }
+    
+  var modelVersion: ModelVersion { ... }
+}
+```
+
+Implementing `modelId` was easy, as iTunes already has a unique track id for all song results — so we return that.
+
+However, iTunes does not have a concept of a version for a given song — the only way to know if any song metadata has changed is to look at the metadata itself. No problem! If your server or data source has no concept of version, Pilot has a handy `ModelVersionMixer` class that lets you combine important metadata into an efficient hashed value. For `Song`, we would want to mix in all the metadata to contribute to the hash. Then if any metadata changes, the version will change:
+
+```swift    
+  var modelVersion: ModelVersion {
+    var mixer = ModelVersionMixer()
+    mixer.mix(artistId)
+    mixer.mix(collectionId)
+    mixer.mix(trackId)
```

Comment:
I don't think there's any benefit to having `trackId` in the `modelVersion` since it's already in the `modelId`.